Lib/test/test_http_cookiejar.py

```diff
@@ -860,12 +860,24 @@ def test_is_HDN(self):
         self.assertTrue(is_HDN("foo.bar.com"))
         self.assertTrue(is_HDN("1foo2.3bar4.5com"))
         self.assertFalse(is_HDN("192.168.1.1"))
+        self.assertFalse(is_HDN("[::1]"))
+        self.assertFalse(is_HDN("[2001:db8:85a3::8a2e:370:7334]"))
         self.assertFalse(is_HDN(""))
         self.assertFalse(is_HDN("."))
         self.assertFalse(is_HDN(".foo.bar.com"))
         self.assertFalse(is_HDN("..foo"))
         self.assertFalse(is_HDN("foo."))
 
+    def test_is_ip_like_hostname(self):
+        self.assertTrue(is_ip_like_hostname('[::1]'))
+        self.assertTrue(is_ip_like_hostname('[2001:db8:85a3::8a2e:370:7334]'))
+        self.assertTrue(is_ip_like_hostname('192.168.0.1'))
+
+        self.assertFalse(is_ip_like_hostname('256.256.256.256'))
```

Comment:
```suggestion
        self.assertFalse(is_ip_like_hostname('::1'))
        self.assertFalse(is_ip_like_hostname('256.256.256.256'))
```

AFAICT, ::1 is now rejected but not [::1].